src/Mutator/Extensions/MBString.php

```diff
@@ -0,0 +1,199 @@
+            'mb_substr' => $this->mapFunctionAndRemoveExtraArgs('substr', 3),
+            'mb_convert_case' => $this->mapConvertCase(),
+        ];
+
+        $functionsToRemove = \array_filter($functionsMap, static function ($isOn) {
+            return !$isOn;
+        });
+
+        $this->converters = \array_diff_key($converters, $functionsToRemove);
+    }
+
+    private function mapFunction(string $newFunctionName): callable
+    {
+        return function (Node\Expr\FuncCall $node) use ($newFunctionName): Generator {
+            yield $this->mapFunctionCall($node, $newFunctionName, $node->args);
+        };
+    }
+
+    private function mapFunctionAndRemoveExtraArgs(string $newFunctionName, int $argsAtMost): callable
+    {
+        return function (Node\Expr\FuncCall $node) use ($newFunctionName, $argsAtMost): Generator {
+            yield $this->mapFunctionCall($node, $newFunctionName, \array_slice($node->args, 0, $argsAtMost));
+        };
+    }
+
+    private function mapConvertCase(): callable
+    {
+        return function (Node\Expr\FuncCall $node): Generator {
+            $modeValue = $this->getConvertCaseModeValue($node);
+
+            if ($modeValue === null) {
+                return;
+            }
+
+            $functionName = $this->getConvertCaseFunctionName($modeValue);
+
+            if ($functionName === null) {
+                return;
+            }
+
+            yield $this->mapFunctionCall($node, $functionName, [$node->args[0]]);
+        };
+    }
+
+    private function getConvertCaseModeValue(Node\Expr\FuncCall $node): ?int
+    {
+        if (\count($node->args) < 2) {
+            return null;
+        }
+
+        $mode = $node->args[1]->value;
+
+        if ($mode instanceof Node\Scalar\LNumber) {
+            return $mode->value;
+        }
+
+        if ($mode instanceof Node\Expr\ConstFetch) {
+            return \constant($mode->name->toString());
+        }
+
+        return null;
+    }
+
+    private function getConvertCaseFunctionName(int $mode): ?string
+    {
+        if ($this->isInMbCaseMode($mode, 'MB_CASE_UPPER', 'MB_CASE_UPPER_SIMPLE')) {
+            return 'strtoupper';
+        }
+
+        if ($this->isInMbCaseMode($mode, 'MB_CASE_LOWER', 'MB_CASE_LOWER_SIMPLE', 'MB_CASE_FOLD', 'MB_CASE_FOLD_SIMPLE')) {
+            return 'strtolower';
+        }
+
+        if ($this->isInMbCaseMode($mode, 'MB_CASE_TITLE', 'MB_CASE_TITLE_SIMPLE')) {
+            return 'ucwords';
+        }
+
+        return null;
+    }
+
+    private function isInMbCaseMode(int $mode, string ...$cases): bool
+    {
+        foreach ($cases as $constant) {
+            if (\defined($constant) && \constant($constant) === $mode) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private function getFunctionName(Node $node): ?string
+    {
+        if (!$node instanceof Node\Expr\FuncCall || !$node->name instanceof Node\Name) {
+            return null;
```

Comment:
This check is needed only once, in `mutatesNode`.  When it will be in  `mutatesNode`, it will be immediately obvious what this mutator is targeted to. Also one less null returned.

Reply:
I removed `getFunctionName` altogether

Reply:
That's even better!